From c955f3ea8921c91b09ec3e00aee829d20146647c Mon Sep 17 00:00:00 2001
From: Yunsik Jung <yunsik.jung@lge.com>
Date: Mon, 16 Jun 2025 20:25:04 +0900
Subject: [PATCH] [build][upstreamable] 138.0.7170.0~1: V8: Rust: Disable Temporal API

:Release Notes:
Disabled building V8 Rust-depended Temporal API to fix Neva build after rolling
to 138.0.7170.0~1

:Detailed Notes:
In the upstream V8 CL
https://chromium-review.googlesource.com/c/v8/v8/+/6514659, which is rolled in
Chromium in the upstream CL https://crrev.com/c/6522336, added the build
argument 'v8_enable_temporal_support' which is to support Temporal API in v8.

This feature has dependency on the temporal_rs library and built with
Rust but Neva doesn't support Rust build yet.

Thus, to fix build error, 'v8_enable_temporal_support' set to false if
'enable_rust=false'.

Without this patch the build fails with error:
| ERROR at //build/rust/rust_target.gni:41:3: Assertion failed.
|   assert(enable_rust)
|   ^-----
| See //build/rust/rust_static_library.gni:214:3: whence it was called.
|   rust_target(_target_name) {
|   ^--------------------------
| See //build/rust/cargo_crate.gni:229:3: whence it was called.
|   target(_target_type, target_name) {
|   ^----------------------------------
| See //third_party/rust/temporal_capi/v0_0/BUILD.gn:11:1: whence it was called.
| cargo_crate("lib") {
| ^-------------------
| See //v8/BUILD.gn:2330:7: which caused the file to be included.
|       "//third_party/rust/temporal_capi/v0_0:lib",

:Issues Addressed:
[NEVA-10393] Upgrade neva-main-rolling up to 7170 BUILD
---
 gni/v8.gni | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/gni/v8.gni b/gni/v8.gni
index d198859576a..d5f093ddeb7 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -18,6 +18,11 @@ if (is_ios) {
   import("//build/config/apple/mobile_config.gni")  # For `target_platform`.
 }
 
+# TODO(neva_rust): Remove this workaround after Neva supports Rust.
+if (build_with_chromium) {
+  import("//neva/neva.gni")
+}
+
 declare_args() {
   # Includes files needed for correctness fuzzing.
   v8_correctness_fuzzer = false
@@ -61,8 +66,10 @@ declare_args() {
   # is still not accessible unless --harmony-temporal is enabled at runtime)
   #
   # Furthermore, some architectures don't have Rust toolchains in Chromium
+  # TODO(neva_rust): Remove this workaround after Neva supports Rust.
   v8_enable_temporal_support = !(defined(build_with_node) && build_with_node) &&
                                target_cpu != "ppc64" && target_cpu != "s390x"
+                               && !(build_with_chromium && disable_rust_for_neva)
 
   # Use static libraries instead of source_sets.
   v8_static_library = false
-- 
2.51.2

